var/spack/repos/builtin/packages/elk/package.py

```diff
@@ -34,34 +35,29 @@ def configure(self, spec):
         # Dictionary of configuration options
         config = {
             'MAKE':      'make',
-            'F90':       join_path(spack.build_env_path, 'f90'),
-            'F77':       join_path(spack.build_env_path, 'f77'),
-            'AR':        'ar',
-            'LIB_FFT':   'fftlib.a',
-            'SRC_MPI':   'mpi_stub.f90',
-            'SRC_OMP':   'omp_stub.f90',
-            'SRC_libxc': 'libxcifc_stub.f90',
-            'SRC_FFT':   'zfftifc.f90'
+            'AR':        'ar'
         }
 
         # Compiler-specific flags
         flags = ''
         if   self.compiler.name == 'intel':
-            flags = '-O3 -ip -unroll -no-prec-div -openmp'
+            flags = '-O3 -ip -unroll -no-prec-div'
         elif self.compiler.name == 'gcc':
-            flags = '-O3 -ffast-math -funroll-loops -fopenmp'
+            flags = '-O3 -ffast-math -funroll-loops'
         elif self.compiler.name == 'pgi':
-            flags = '-O3 -mp -lpthread'
+            flags = '-O3 -lpthread'
         elif self.compiler.name == 'g95':
             flags = '-O3 -fno-second-underscore'
         elif self.compiler.name == 'nag':
             flags = '-O4 -kind=byte -dusty -dcfuns'
         elif self.compiler.name == 'xl':
-            flags = '-O3 -qsmp=omp'
+            flags = '-O3'
         config['F90_OPTS'] = flags
         config['F77_OPTS'] = flags
 
         # BLAS/LAPACK support
+        # Note: BLAS/LAPACK must be compiled with OpenMP support
+        # if the +openmp variant is chosen
```

Comment:
you may try to use features from #657, namely `self.spec.blas_shared_lib` and alike.